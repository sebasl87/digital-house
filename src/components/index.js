export { default as TitleSection } from "./TitleSection";
export { default as TotalBox } from "./TotalBox";
