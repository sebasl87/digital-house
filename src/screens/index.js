export { default as HomeScreen } from "./HomeScreen";
export { default as ProductDetail } from "./ProductDetail";
