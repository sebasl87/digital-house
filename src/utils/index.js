import { MONTHS } from "../constants";

export const getTotal = (arr) => {
  const sum = arr
    .filter((mov) => mov.is_redemption)
    .reduce(function (prev, current) {
      return prev + +current.points;
    }, 0);
  const rest = arr
    .filter((mov) => mov.is_redemption === false)
    .reduce(function (prev, current) {
      return prev + +current.points;
    }, 0);
  const result = (sum - rest).toLocaleString("en");
  return result;
};

export const formatDate = (fecha) => {
  const dateFormat =
    fecha.slice(8, 10) +
    " de " +
    MONTHS[fecha.slice(5, 7)] +
    ", " +
    fecha.slice(0, 4);

  return dateFormat;
};
