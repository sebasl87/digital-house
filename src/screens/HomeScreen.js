import React, { useContext } from "react";
import { View, Text, ScrollView } from "react-native";
import mainContext from "../../context/mainContext";

import { styles } from "../../styles";
import { TitleSection, TotalBox } from "../components";

function HomeScreen() {
  const { movements, points } = useContext(mainContext);
  // console.log(points.toLocaleString("en"));
  return (
    <ScrollView
      contentContainerStyle={{
        paddingVertical: 60,
      }}
    >
      <View
        style={{
          flex: 1,
          alignItems: "flex-start",
          justifyContent: "center",
          padding: 20,
        }}
      >
        <View>
          <Text style={styles.titleWelcomme}>Bienvenido de vuelta!</Text>
          <Text>Ruben Rodriguez</Text>
        </View>
        <TitleSection text="TUS PUNTOS" />
        <TotalBox points={points} />
      </View>
    </ScrollView>
  );
}

export default HomeScreen;
