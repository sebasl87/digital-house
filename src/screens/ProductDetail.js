import React, { useContext } from "react";
import { View, ScrollView, Image } from "react-native";
import mainContext from "../../context/mainContext";
import { CardDetail } from "../components";

function ProductDetail() {
  const { movements, points, styles } = useContext(mainContext);
  return (
    <ScrollView
      contentContainerStyle={{
        padding: 20,
        width: "100%",
      }}
    >
      <View style={styles.containerImg}>
        <Image
          source={{ uri: "https://loremflickr.com/640/480/city" }}
          style={{ width: "100%", height: 350, borderRadius: 10 }}
        />
      </View>
      <CardDetail />
    </ScrollView>
  );
}

export default ProductDetail;
