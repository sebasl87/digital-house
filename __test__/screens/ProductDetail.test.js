import "react-native";
import React from "react";
import renderer from "react-test-renderer";
import { ProductDetail } from "../../src/screens";
import { render, fireEvent } from "react-native-testing-library";

const mockHistoryGoBack = jest.fn();

jest.mock("@react-navigation/native", () => ({
  ...jest.requireActual("@react-navigation/native"),
  useNavigation: () => ({
    goBack: mockHistoryGoBack,
  }),
  useRoute: () => ({
    params: {
      date: "2022-12-09T06:34:25.607Z",
      product: "Handmade Metal Shoes",
      points: 16434,
      image: "https://loremflickr.com/640/480/transport",
      isRedemption: false,
    },
  }),
}));

jest.mock("react", () => ({
  ...jest.requireActual("react"),
  useContext: () => ({
    styles: { containerImg: {}, button: {}, monthBlueBox: {} },
  }),
}));

describe("ProductDetail screen", () => {
  it("should render successfully with mock params", () => {
    const productDetail = renderer.create(<ProductDetail />).toJSON();
    expect(productDetail).toMatchSnapshot();
  });

  it("should be redirect on press", () => {
    const { getByTestId } = render(<ProductDetail />);
    fireEvent.press(getByTestId("touchableGoBackTolist"));

    expect(mockHistoryGoBack).toHaveBeenCalled();
  });
});
