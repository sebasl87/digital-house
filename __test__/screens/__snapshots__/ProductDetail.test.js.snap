// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductDetail screen should render successfully with mock params 1`] = `
<RCTScrollView
  contentContainerStyle={
    {
      "padding": 20,
      "width": "100%",
    }
  }
>
  <View>
    <View
      style={{}}
    >
      <Image
        source={
          {
            "uri": "https://loremflickr.com/640/480/transport",
          }
        }
        style={
          {
            "borderRadius": 10,
            "height": 350,
            "width": "100%",
          }
        }
      />
    </View>
    <View
      style={
        {
          "flex": 1,
          "justifyContent": "center",
          "width": "100%",
        }
      }
    >
      <View
        style={
          {
            "alignItems": "flex-start",
            "marginVertical": 20,
            "width": "100%",
          }
        }
      >
        <Text
          style={
            {
              "color": "#9B9898",
              "fontFamily": "Avenir",
              "fontSize": 14,
              "fontStyle": "normal",
              "fontWeight": "800",
              "lineHeight": 19,
            }
          }
        >
          Detalles del producto
        </Text>
      </View>
      <Text
        style={
          [
            {
              "color": "#FFF",
              "fontFamily": "Avenir",
              "fontSize": 16,
              "fontStyle": "normal",
              "fontWeight": "800",
              "lineHeight": 22,
            },
            {
              "color": "#000",
            },
          ]
        }
      >
        Comprado el 
        09 de Diciembre, 2022
      </Text>
      <View
        style={
          {
            "alignItems": "flex-start",
            "marginVertical": 20,
            "width": "100%",
          }
        }
      >
        <Text
          style={
            {
              "color": "#9B9898",
              "fontFamily": "Avenir",
              "fontSize": 14,
              "fontStyle": "normal",
              "fontWeight": "800",
              "lineHeight": 19,
            }
          }
        >
          Con esta compra acumulaste:
        </Text>
      </View>
      <Text
        style={
          [
            {
              "color": "#FFF",
              "fontFamily": "Avenir",
              "fontSize": 16,
              "fontStyle": "normal",
              "fontWeight": "800",
              "lineHeight": 22,
            },
            {
              "color": "#000",
              "fontSize": 24,
              "lineHeight": 33,
            },
          ]
        }
      >
        16434
         puntos
      </Text>
    </View>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "opacity": 1,
        }
      }
      testID="touchableGoBackTolist"
    >
      <Text
        style={{}}
      >
        Aceptar
      </Text>
    </View>
  </View>
</RCTScrollView>
`;
